l10n_ch_pain_base/models/account_payment_order.py

```diff
@@ -68,7 +67,8 @@ def generate_party_agent(
             if bank_line.local_instrument == 'CH01':
                 # Don't set the creditor agent on ISR/CH01 payments
                 return True
-            elif not partner_bank.bank_bic:
+            elif not partner_bank.bank_bic and \
+                    any(self.mapped("payment_method_id.bic_required")):
```

Comment:
No need for an additional field

```suggestion
            elif not partner_bank.bank_bic and not partner_bank.type == "iban":
```